misc/architecture-model-generator/src/main/java/io/ballerina/architecturemodelgenerator/generators/entity/EntityModelGenerator.java

```diff
@@ -119,9 +119,12 @@ private Entity getType(RecordTypeSymbol recordTypeSymbol, RecordTypeDescriptorNo
         for (RecordFieldSymbol recordFieldSymbol : recordFieldSymbolMap.values()) {
             RecordFieldNode recordFieldNode = null;
             if (recordFieldSymbol.getName().isPresent() && recordNode != null) {
-                recordFieldNode = (RecordFieldNode) recordNode.fields().stream()
-                        .filter(node -> ((RecordFieldNode) node).fieldName().text().equals(recordFieldSymbol.getName()
-                                .get())).findFirst().orElse(null);
+                recordFieldNode = (RecordFieldNode) recordNode.fields().stream().filter(node -> {
+                    if (node.kind().equals(SyntaxKind.RECORD_FIELD)) {
+                        return ((RecordFieldNode) node).fieldName().text().equals(recordFieldSymbol.getName().get());
+                    }
+                    return false;
```

Comment:
shall we use lambda expressions with `&&` instead of using if statements (which will simplify the logic)?

```suggestion
                recordFieldNode = (RecordFieldNode) recordNode.fields().stream().filter(node -> 
                    node.kind().equals(SyntaxKind.RECORD_FIELD) && ((RecordFieldNode) node).fieldName().text().equals(recordFieldSymbol.getName().get());
```